addons/populate/functions/fnc_spawnBuildingUnit.sqf

```diff
@@ -0,0 +1,95 @@
+#include "script_component.hpp"
+/*
+    Author:
+        Bridg
+
+    Description:
+        Spawns a unit at the passed position
+
+    Parameters:
+        0: _spawnPositions <ARRAY>
+        1: _numberOfUnitsToSpawn <NUMBER>
+        2: _numberOfPositionsToOccupy <NUMBER>
+        3: _side <SIDE>
+        4: _module <MODULE>
+        5: _unitPoolArray <ARRAY>
+        6: _statics <ARRAY>
+
+    Return value:
+        Nothing
+*/
+
+params ["_spawnPositions", "_numberOfUnitsToSpawn", "_numberOfPositionsToOccupy", "_side", "_module", "_unitPoolArray", "_statics"];
+
+private _group = grpNull;
+private _currenGrouptUnitCount = 0;
+
+[{
+    params ["_args", "_idPFH"];
+    _args params ["_spawnPositions", "_numberOfUnitsToSpawn", "_numberOfPositionsToOccupy", "_side", "_module", "_unitPoolArray", "_currenGrouptUnitCount", "_group", "_statics"];
+
+    private _pos = [0,0,0];
+    private _dir = -1;
+
+    if (_numberOfPositionsToOccupy <= 1) exitWith {
+        [_idPFH] call cba_fnc_removePerFrameHandler;
+        if (_numberOfUnitsToSpawn > 0) then {
+            [_numberOfUnitsToSpawn, _side, _module, _unitPoolArray] call FUNC(createPatrols);
+        ;}
+    };
+
+    // create a group if the _currenGrouptUnitCount is == 0
+    if (_currenGrouptUnitCount isEqualTo 0) then {
+        _group = createGroup _side;
+    };
+
+    private _unitType = selectRandom _unitPoolArray;
+    private _unit = _group createUnit [_unitType, [0,0,0], [], 0, "NONE"];
+
+    _currenGrouptUnitCount = _currenGrouptUnitCount + 1;
+    _numberOfUnitsToSpawn = _numberOfUnitsToSpawn - 1;
+
+    // do statics first - use exit with to stop spawning of other units
+    if (_statics isNotEqualTo []) exitWith {
+        private _static = selectRandom _statics;
+        _unit moveInAny _static;
+        _statics deleteAt (_statics findIf {_x isEqualTo _static});
+        _numberOfPositionsToOccupy = _numberOfPositionsToOccupy - 1;
+
+        _args set [1, _numberOfUnitsToSpawn];
+        _args set [2, _numberOfPositionsToOccupy];
+        _args set [6, _currenGrouptUnitCount];
+        _args set [7, _group];
+        _args set [8, _statics];
+    };
+
+    private _spawnPos = selectRandom _spawnPositions;
+    if (typeName _spawnPos == "OBJECT") then { // handle cba positions as theyre objects
+        _pos = getPos _spawnPos;
+        _dir = getDir _spawnPos;
+    } else {
+        _pos = _spawnPos;
+    };
+
+    TRACE_4("", _pos, _dir, typeName _spawnPos, _spawnPos);
+
+    _unit setPos _pos;
+    _unit setDir _dir; // groups will follow the group leaders dir most of the time
+    _spawnPositions deleteAt (_spawnPositions findIf {_x isEqualTo _spawnPos});
+     _numberOfPositionsToOccupy = _numberOfPositionsToOccupy - 1;
+
+    _unit disableAI "PATH";
+
+    if (_currenGrouptUnitCount == 10) then {
```

Comment:
Why groups of 10?

Reply:
Complete arbitrary. Theres a chance this will be used to quickly populate large areas, didn't wanna all 50 guys in one group. You thinking of some kind of logic instead?